### DISCUSSION TITLE
I can't understand how to use Python CLI

### DISCUSSION THREAD

**Discussion by RyokoNod** [dandi/helpdesk#15] [Q&A] [answered]
Hello, this is a very basic question but I can't figure it out - pretty new to doing things with Python.

I am trying to download two dandisets on my MacOSX. I think I installed the Python CLI correctly using `pip install dandi`. However, I don't understand how to use the command `dandi download https://identifiers.org/DANDI:blahblah`.

Using it directly in my terminal, I get the error `command not found: dandi`. Typing in `python3 -m dandi`, I get the error `'dandi' is a package and cannot be directly executed`. What am I doing wrong?

**Comment by djarecka** [comment, marked as answer]
`dandi` command should work if the package is installed correctly. Perhaps you could create a new conda environment and check if it works:

```
conda create -n dandi python=3.8
conda activate dandi
pip install dandi
dandi --help
```

**Reply by RyokoNod** [reply]
Thank you @djarecka, this worked beautifully, and now I have all the NWB I need for analysis!

**Comment by waxlamp** [comment]
Hi @RyokoNod! Sorry you are running into trouble.

If you run `pip freeze`, do you see `dandi` in the output? That is a good way to ensure that a given package is really installed.

If you do see it there, then I would expect the `dandi` executable to be found at `/usr/bin/dandi` or `/usr/local/bin/dandi` (I am not super familiar with how Pip and Python on OSX behaves).

In a pinch, you can also create a virtual environment using @djarecka's suggestion. You can also do so using the `virtualenv` program if Conda is not available:

```
virtualenv venv       # creates a virtual environment in a local venv directory
. ./venv/bin/actiavte # activates the environment in your shell
pip install dandi
dandi --help
```

Let us know what happens.

**Reply by RyokoNod** [reply]
Hi @waxlamp, thank you for your advice. This time @djarecka's advice worked better because OSX doesn't seem to store a `dandi` executable in `/usr/bin/dandi` or `/usr/local/bin/dandi`. Instead it stores `dandi` as a library in `/Users/my_username/opt/anaconda3/lib/python3.8/site-packages`, which means `dandi` is a directory with a bunch of `.py` files inside. 

Of course, this is in my case so it might not apply to all OSX users. It just might come in handy if there are other OSX users who need help.

**Comment by yarikoptic** [comment]
FWIW: If using conda, `conda install -c conda-forge dandi` should be sufficient (ie no need to pip install it)

**Reply by RyokoNod** [reply]
Oh, hadn't thought of that. But thanks!
